src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -111,6 +111,7 @@ private void ProcessEndpointParameterSource(Endpoint endpoint, ISymbol symbol, I
             // and codegen. Emit a diagnostic when these are encountered to avoid producing buggy code.
             else if (!(SymbolEqualityComparer.Default.Equals(Type, wellKnownTypes.Get(WellKnownType.Microsoft_Extensions_Primitives_StringValues))
                     || Type.SpecialType == SpecialType.System_String
+                    || (IsArray && ElementType.SpecialType == SpecialType.System_String)
                     || TryGetParsability(Type, wellKnownTypes, out var _)
                     || (IsArray && TryGetParsability(ElementType, wellKnownTypes, out var _))))
```

Comment:
I think I understand now why I thought this was an intentional change to emit the diagnostic when I looked at fixing this myself - I misread the braces and thought it was `!a or b or c or d` rather than `!(a or b or c or d)`.

Is it worth adjusting the alignment slightly to improve the readability? Subjective opinion, might just be that **I** misread it and it's clear to others.

```suggestion
            else if (!(SymbolEqualityComparer.Default.Equals(Type, wellKnownTypes.Get(WellKnownType.Microsoft_Extensions_Primitives_StringValues))
                       || Type.SpecialType == SpecialType.System_String
                       || (IsArray && ElementType.SpecialType == SpecialType.System_String)
                       || TryGetParsability(Type, wellKnownTypes, out var _)
                       || (IsArray && TryGetParsability(ElementType, wellKnownTypes, out var _))))
```

Reply:
Good feedback -- I'll do you one better as well and move this to a seperate method.